### PR TITLE
fix(commands): successive scrollIntoView

### PR DESCRIPTION
**Pull request by OBe95**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)
Fixes [#9805](https://github.com/webdriverio/webdriverio/issues/9805)
Subsequent `scrollIntoView` on a given element, is wrongly scrolling the page
It seems that we have 2 issues:
- the `origin` used when scrolling is different than the one used when computing the deltas
- the `deltaX` seems to be wrongly calculated (using `height` instead of `width`)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/webdriverio/src/commands/element/scrollIntoView.ts [hunk 1]
```diff
@@ -56,19 +56,20 @@ export async function scrollIntoView (
...
      */
     if (typeof options === 'boolean' || typeof options.block === 'string' || typeof options.inline === 'string') {
-        const htmlElem = await browser.$('html')
-        const viewport = await htmlElem.getSize()
+        origin = await browser.$('html')
+        const viewport = await origin.getSize()
         const elemSize = await this.getSize()
         if (options === true || (options as ScrollIntoViewOptions).block === 'start') {
-            deltaY += viewport.height - elemSize.height
+            deltaY += viewport.width - elemSize.width
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/element/scrollIntoView.ts:70]: Can you explain how to determine the vertical delta when looking at the `wdith`of the element?

**Review by christian-bromann — COMMENTED** [review]
I wonder if we can have any e2e tests for this, please see the `./e2e` folder, so we can make sure this doesn't cause any regression.

File: packages/webdriverio/src/commands/element/scrollIntoView.ts (hunk 1 above)
Comment (OBe95) [reply packages/webdriverio/src/commands/element/scrollIntoView.ts:70]: Unless I'm missing something, but from my understanding the vertical delta is computed in the `deltaX` variable in the following lines

Here it was computing the horizontal delta but based on the `height` which doesn't make sense

**Comment by OBe95** [comment]
Not sure if it's an automatic message regarding e2e, but I did cover it by scrolling once and getting the `window` scroll state, then doing successive scrolling and checking that the `window` scroll state matches the first one (assuming here that one scroll works just fine)

File: packages/webdriverio/src/commands/element/scrollIntoView.ts (hunk 1 above)
Comment (OBe95) [reply packages/webdriverio/src/commands/element/scrollIntoView.ts:70]: My bad, I meant to update the `deltaX`
They were both based the `height`, I will push an update soon

**Comment by christian-bromann** [comment]
I would suggest the following method of testing this:

- run `elem.scrollIntoView()`
- get scroll position
- run web API: `browser.execute((elem) => elem.scrollIntoView(), [elem])`
- compare if both are the same

I think this better tells us whether our command behaves the same way as the web API.

**Comment by OBe95** [comment]
@christian-bromann I pushed an update to handle all options, add e2es and update the snapshots for existing unit tests

**Review by christian-bromann — APPROVED** [review]
This is awesome, thanks so much 👍

**Comment by christian-bromann** [comment]
> ** Test / e2e-windows (pull_request) ** Failing after 25m
> ** Test / e2e-mac (pull_request) ** Failing after 9m

Seems both fail by `5px` different, I will go ahead and merge this and fix the test failure in `main`

**Comment by niklasschaeffer** [comment]
This fix has been leading to an weird debug message

```
ERROR webdriver: Request failed with status 400 due to invalid argument: invalid argument
[0-0] from invalid argument: 'delta x' must be an int
[0-0]   (Session info: chrome=111.0.5563.64)
```

**Comment by km9134** [comment]
Has anyone noticed any issues when using the block parameter in this function? When i call `scrollIntoView({ behavior: 'smooth',  block: 'top' })` I am presented with an error `TypeError: Cannot read properties of undefined (reading 'y')`

Notably if i use `scrollIntoView({ behavior: 'smooth',  block: 'center' })` i am not presented with any issues

**Comment by christian-bromann** [comment]
@niklasschaeffer thanks for reporting, mind raising a PR with changes that would round the `deltaX` and `deltaY` values?

**Comment by OBe95** [comment]
Let me know @niklasschaeffer if you want me to take care of it 😄

**Comment by niklasschaeffer** [comment]
@OBe95 @christian-bromann there should be a pr now. dont know if it actually fixes the problem :D

**Pull request merged**